### PR TITLE
Improve and extend the security checklist

### PR DESCRIPTION
**Pull request by glye**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | All
| Edition       | All

Reorganised the checklist to make it more intuitive, and added more advice:

- Moved Ibexa DXP to the top, above Symfony and PHP.
- Added Web server and Database sections.
- Moved some items from the Ibexa DXP section to the new sections.
- Added Domain section.
- Added advice on TLS, HSTS, DNSSEC, CAA, and domain update protection.

Preview: https://ez-systems-developer-documentation--2565.com.readthedocs.build/en/2565/infrastructure_and_maintenance/security/security_checklist/

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] ~Code samples are working~ (no changes to code samples)
- [x] ~PHP code samples have been fixed with PHP CS fixer~ (no changes to code samples)
- [x] ~Added link to this PR in relevant JIRA ticket or code PR~

### CODE REVIEW

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:292]: Oxford comma 🧐 
```suggestion
- Ensure any servers, services, ports, and virtual hosts that were opened for testing purposes are shut down before going live.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:251]: Vale suggestion:
```suggestion
It protects against DNS poisoning attacks, which is when an attacker manipulates the reponses to DNS requests with the goal of directing users to an IP address the attacker controls.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:258]: Vale suggestion:
```suggestion
You can enable this protection at your domain registrar's site.
```

**Review by glye — COMMENTED** [review]
Self-review.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:251]: Pending reviewers approval.

**Comment by reithor** [comment]
regarding 'powered by' we should probably give explicit instructions, like:

```yaml

ibexa_system_info:
  system_info:
    powered_by:
      # major => v4 || minor => v4.6 || none
      release: major
      # true || false
      enabled: true
      
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (bdunogier) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:171]: Repetition of the above.

Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:171]: Huh, that's a flaw from long ago apparently, but let's fix it. Seems we can simply delete the line.
```suggestion
```

**Review by reithor — COMMENTED** [review]
Looks good to me (as far my technical knowledge goes here).
One thing regarding "Security headers":
Probably we should review our cloud vcl for varnish and fastly, if we are following our own guidelines?

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (adriendupuis) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:15]: Maybe, in the paragraph below I would remind that the user list must be kept up-to-date, anyone leaving the organization should be disable without delay, or even one changing job in the organization, should have role updated.

**Comment by glye** [comment]
@reithor I and others have held back on the security headers due to the need to adapt them to the site in question. Generally there isn't a "safe setting" that can be applied for everyone. But I have been reconsidering. We could change vcl and vhost templates, they won't affect existing sites. We could set at least Strict-Transport-Security, and X-Content-Type-Options to nosniff. For the admin ui we could set all of them.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (adriendupuis) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:154]: Maybe something about `.env.local` containing all the secrets, passwords, and sensible data, even `.env.local` added to `.gitignore`.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (adriendupuis) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:294]: So, what at the rights to give to web app users to have the needed ones and only the needed ones? (see #2356)

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:15]: "Carefully selected admin users"? Good point about the internal users.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:154]: Good stuff. Also, I'll need to modify the related "Secure secrets" section.

Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:154]:
```suggestion
- The secret must be secured against unwanted access. Don't commit the value to a version control system. There are several ways of handling it, like with enviroment variables or files like `.env.local`. Files are considered more secure. If you store the secrets in files, make sure to add those files to `.gitignore` or similar, so they will never be committed to version control systems.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md]:
```suggestion
Don't expose the dev mode on the internet, otherwise you may disclose things like `phpinfo`, environment variables, and more.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:172]:
```suggestion
For more information about securing Symfony-based systems, see [Authentication and authorisation]([[= symfony_doc =]]/security.html), [more on this subject]([[= symfony_doc =]]/security.html#learn-more), and [secrets management system]([[= symfony_doc =]]/configuration/secrets.html), all from Symfony.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:180]:
```suggestion
On production sites, this should be set to 1 (enabled) to ensure that stack traces don't include arguments passed to functions.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:182]:
```suggestion
You should also make sure that no stack trace is ever visible to end users of production sites. Visible arguments are unsafe even if the stack traces only show up in log files.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:189]: These? Do you mean "the following"?

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:208]:
```suggestion
Make sure that the web server blocks the execution of PHP files and other scripts in the `var` directory.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:240]:
```suggestion
Removing them means that attackers can't attempt to force other users to use weak ciphers and eavesdrop on their communications.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:241]:
```suggestion
TLS 1.3 is as of December 2024 [supported by ca. 97% of global internet users](https://caniuse.com/tls1-3).
```
I'd put "As of December 2024," in the first place in the sentence.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:244]:
```suggestion
When using [[= product_name_cloud =]], you can [set the minimum TLS version in `.platform/routes.yaml`](https://docs.platform.sh/define-routes/https.html#enforce-tls-13).
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:251]:
```suggestion
Make sure to also include subdomains by means of the `includeSubDomains` setting.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:253]:
```suggestion
When using [[= product_name_cloud =]], you can [configure HSTS in `.platform/routes.yaml`](https://docs.platform.sh/define-routes/https.html#enable-http-strict-transport-security-hsts).
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:262]:
```suggestion
[Read more on DNSSEC on ICANN's website](https://www.icann.org/resources/pages/dnssec-what-is-it-why-important-2019-03-05-en).
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (dabrt) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:287]:
```suggestion
Secure the database access with strong passwords, keys, firewall, encryption in transit, encryption at rest, and so on, as needed.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (adriendupuis) [inline docs/infrastructure_and_maintenance/security/security_checklist.md:161]: To have the copy to clipboard button:

```suggestion
    ```bash
    php -r "print bin2hex(random_bytes(32));"
    ```
```

**Review by mnocon — COMMENTED** [review]
Some smaller nitpicks - the HSTS header with Varnish being the most important one

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md]: @dabrt This was resolved earlier in a different way, let me know if you still want it changed.

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (mnocon) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:246]: Or actually maybe I misunderstood it and it's ok to enforce HSTS, because the purge request is sent from the back office to Varnish, not the other way🤔 I'm not sure right now

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:189]:
```suggestion
The following `php.ini` configuration values should be used on production sites.
```

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:248]: Nice catch. We can say that this should apply between the client and the public-facing proxy, if any, otherwise the webserver. I'll come up with a sentence or warning box about it.

Comment (mnocon) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:248]: It's safer for me to ping @vidarl here, he knows better what's the recommended approach 😄

Comment (vidarl) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:248]: These settings are related to what happends between client and public-facing proxy, yes.

Even with both https and HSTS enabled, varnish and app container will still be reachable to each other over plain http

Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md:249]: Thanks, handled in https://github.com/ibexa/documentation-developer/pull/2565/commits/7eb8d61d5482fdcf3f1dafc3c0534b4a847bc230

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md]: @adriendupuis Keeping users up to date handled in https://github.com/ibexa/documentation-developer/pull/2565/commits/377d9c8f231f0ba196ea5fc5277bb02766a37365

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md]: Merged "Secure secrets" into this section: https://github.com/ibexa/documentation-developer/pull/2565/commits/6941f052cab1fe9d8f56c551a56a2feae940ff4d

File: docs/infrastructure_and_maintenance/security/security_checklist.md
Comment (glye) [reply docs/infrastructure_and_maintenance/security/security_checklist.md]: @adriendupuis This isn't documented AFAIK. Ref. the doc that only says "Before executing the command make sure that the database user has sufficient permissions."
https://doc.ibexa.co/en/latest/getting_started/install_ibexa_dxp/#create-a-database

Incidentally, I found that the next paragraph doesn't belong there. PR to move it: https://github.com/ibexa/documentation-developer/pull/2569

Anyway, I removed the section, and instead added a new Monitor section at the end, about monitoring the various part of the system. See https://github.com/ibexa/documentation-developer/pull/2565/commits/26d5a78bbefe56e76027093f57f525353420657f

**Comment by glye** [comment]
Thanks for all the reviews, very useful. I believe I have answered all comments and applied all suggestions now, and it's looking good. Please have another look, if you had concerns earlier.

**Review by dabrt — APPROVED** [review]

**Review by mnocon — APPROVED** [review]

**Review by adriendupuis — APPROVED** [review]

**Review by reithor — APPROVED** [review]
looks good to me now!

**Review by vidarl — APPROVED** [review]

**Review by bdunogier — APPROVED** [review]

**Pull request merged**
